plane/src/drone/docker/commands.rs

```diff
@@ -231,3 +291,152 @@ pub async fn run_container(
+            key: KeyConfig {
+                name: key.to_string(),
+                ..Default::default()
+            },
+            deadlines: KeyDeadlines {
+                renew_at: LoggableTime(UNIX_EPOCH.into()),
+                soft_terminate_at: LoggableTime(UNIX_EPOCH.into()),
+                hard_terminate_at: LoggableTime(UNIX_EPOCH.into()),
+            },
+            token: Default::default(),
+        });
+
+        let mut exec_config = ExecutorConfig::from_image_with_defaults(String::default());
+        exec_config.mount = mount;
+
+        get_container_config_from_executor_config(
+            &backend_name,
+            exec_config,
+            None,
+            acquired_key.as_ref(),
+            None,
+            None,
+            Some(&PathBuf::from(mount_base)),
+        )
+    }
+
+    // Test basic mount options
+
+    #[test]
+    fn test_mount_true() {
+        let mount_base = "/mnt/my-nfs";
+        let mount = Some(Mount::Bool(true));
+        let expected_binds = Some(vec![format!("/mnt/my-nfs/key:{}", PLANE_DATA_DIR)]);
+
+        let config = get_container_config_from_mount(mount_base, mount)
+            .unwrap()
+            .host_config
+            .unwrap()
+            .binds;
+
+        assert_eq!(config, expected_binds);
+    }
+
+    #[test]
+    fn test_mount_path() {
+        let mount_base = "/mnt/my-nfs";
+        let mount = Some(Mount::Path(PathBuf::from("custom-mount")));
+        let expected_binds = Some(vec![format!("/mnt/my-nfs/custom-mount:{}", PLANE_DATA_DIR)]);
+
+        let config = get_container_config_from_mount(mount_base, mount)
+            .unwrap()
+            .host_config
+            .unwrap()
+            .binds;
+
+        assert_eq!(config, expected_binds);
+    }
+
+    #[test]
+    fn test_mount_false() {
+        let mount_base = "/mnt/my-nfs";
+        let mount = Some(Mount::Bool(false));
+        let expected_binds = None;
+
+        let config = get_container_config_from_mount(mount_base, mount)
+            .unwrap()
+            .host_config
+            .unwrap()
+            .binds;
+
+        assert_eq!(config, expected_binds);
+    }
+
+    #[test]
+    fn test_mount_none() {
+        let mount_base = "/mnt/my-nfs";
+        let mount = None;
+        let expected_binds = None;
+
+        let config = get_container_config_from_mount(mount_base, mount)
+            .unwrap()
+            .host_config
+            .unwrap()
+            .binds;
+
+        assert_eq!(config, expected_binds);
+    }
+
+    // Test invalid mount paths
+
+    #[test]
+    #[should_panic(expected = "not under the mount base")]
+    fn test_mount_path_invalid_parent() {
+        let mount_base = "/mnt/my-nfs";
+        let mount = Some(Mount::Path(PathBuf::from("..")));
+
+        let _config = get_container_config_from_mount(mount_base, mount)
+            .unwrap()
+            .host_config
+            .unwrap()
+            .binds;
```

Comment:
I like to reserve `should_panic` for cases where the implementation under test itself panics, rather than have the test code itself promote an `Err` into a `panic` with `unwrap()`.

In this case, something like this should work:

```rust
let err = get_container_config_from_mount(..).unwrap_err();
assert!(err.to_string(...).contains("not under the mount base"));
```